app/src/main/java/com/mealsmadeeasy/ui/home/glance/WeekAtAGlanceFragment.kt

```diff
@@ -42,15 +43,19 @@ class WeekAtAGlanceFragment : BaseFragment() {
                 .subscribe({ mealPlan ->
                     (rv.adapter as? WeekAtAGlanceAdapter).let {
                         if (it == null) {
-                            rv.adapter = WeekAtAGlanceAdapter(mealPlan = mealPlan) { mealPortion, mealPeriod, date ->
-                                mealStore.removeMealFromMealPlan(mealPortion.meal, date, mealPeriod)
-                                Snackbar.make(root.findViewById<RecyclerView>(R.id.week_at_a_glance_recycler_view),
-                                        R.string.week_at_a_glance_meal_deleted, Snackbar.LENGTH_SHORT)
-                                        .setAction(R.string.week_at_a_glance_undo_delete) {
-                                            mealStore.addMealToMealPlan(mealPortion.meal, date, mealPeriod, mealPortion.servings)
-                                        }
-                                        .show()
-                            }
+                            rv.adapter = WeekAtAGlanceAdapter(mealPlan = mealPlan,
+                                    onDeleteMeal = { mealPortion, mealPeriod, date ->
+                                        mealStore.removeMealFromMealPlan(mealPortion.meal, date, mealPeriod)
+                                        Snackbar.make(root.findViewById<RecyclerView>(R.id.week_at_a_glance_recycler_view),
+                                                R.string.week_at_a_glance_meal_deleted, Snackbar.LENGTH_SHORT)
+                                                .setAction(R.string.week_at_a_glance_undo_delete) {
+                                                    mealStore.addMealToMealPlan(mealPortion.meal, date, mealPeriod, mealPortion.servings)
+                                                }
+                                                .show()
+                                    }, onUpdateMeal = { mealPortion, mealPeriod, date ->
+                                        mealStore.removeMealFromMealPlan(mealPortion.meal, date, mealPeriod)
+                                        mealStore.addMealToMealPlan(mealPortion.meal, date, mealPeriod, mealPortion.servings)
+                            })
```

Comment:
This is getting kinda complicated, so let's pull these two lambdas out into functions and refer to them using method references.